### ISSUE TITLE
Updating branches in Galaxy fork

### ISSUE THREAD

**Issue by marcverhagen** [lappsgrid-incubator/Galaxy#4]
Some notes after Keigh and I looked into how to upgrade the Galaxy instance.

There are two branches `lapps` for lappsgrid.org and `lappsdev` for jetstream. The first is a branch on upstream commit  `5b3debc` and the second on `f9eae90`, none of those have tags so they were probably the head of the master branch at the time of branch creation.

We tried merging the commit with tag `v18.05` into both and got 7 and 10 conflicts respectively. Some conflicts are related to modified Javascript files and some to files that have been deleted upstream. We have to look into whether the modifications on our branches are significant, at the moment we do not even know whether these might perhaps be just whitespace differences.

We tried merging `f9eae90` into `lapps`, which gave a couple of modification errors, but no deleted files. 

All merging is done on local copies, which we will flush before doing it all in a neat way.

Some other thoughts:

- It is probably smart to only merge tagged releases into `lapps` and `lappsdev`.

- We probably should tag the merges on our end, for example, if we merge `v18.05` into `lapps` we should give it a tag `lapps-v18.05`.

- Maybe cull the number of branches in our fork.

- Add a file `lapps-changes.md` that spells out what the differences are and what the `lapps` and `lappsdev` branches are.

**Comment by keighrim** [comment]
Actually, after we talked, I found that [`lappsdev`](417ed0) is [started from](45cd15) marging `f9eae90` and [`lapps`](34e709). This is the diff between those (from `git diff --no-prefix 45cd15 lappsdev`)

https://gist.github.com/keighrim/2f1edcc780af4c78205e355507348e42

**Comment by marcverhagen** [comment]
Looks like they are almost the same and could be used interchangeably and therefore be merged, except perhaps not because of the changes in `scripts/api/common,py` but those look like debugging statements to me.

**Comment by ksuderman** [comment]
We should probably merge lappsdev into lapps and then merge upstream/master into origin/lapps

**Comment by keighrim** [comment]
`v18.05` release is merged to `lapps` branch after `lapps` was merged with `lappsdev`.  See [`merge-2018-07-24`](https://github.com/lappsgrid-incubator/Galaxy/tree/merge-2018-07-24). 
@ksuderman how can I test this version with tools in `galaxy-mods` installed?

**Comment by ksuderman** [comment]
The easiest way is to check out the branch on your local machine and run it there.  Alternatively we can test it on the jetstream instance.

**Comment by keighrim** [comment]
I was able to successfully run galaxy from `merge-2018-07-24` branch after a fix of merge conflict resolution. However, no lapps customization seem to work, as in the screenshot. 

![image](https://user-images.githubusercontent.com/9062727/43370965-7ecddd70-9356-11e8-9303-c7893aa57eca.png)

Any idea..?

**Comment by ksuderman** [comment]
It looks like Galaxy is using the default galaxy.ini file  I will try it on the jetstream server after our call tomorrow.

- Keith

**Comment by keighrim** [comment]
Maybe because I didn't merge [`GalaxyMods` ](https://github.com/lappsgrid-incubator/GalaxyMods)? I looked at the `mods` repository and found several shell scripts, but none of them seemed to be for the merger. Can I just copy everything from `mods` to galaxy (overwriting existing files), or is there a specific merging step?

**Comment by ksuderman** [comment]
Just copy the galaxy.ini and edit for your setup. The galaxy.ini file tells Galaxy where to find tools, plugins, tool_conf.xml, etc.  The galaxy/config/galaxy.ini is not included in source control so updating Galaxy does not overwrite it and this typically only needs to be done once.

**Comment by keighrim** [comment]
After I copied `galaxy.ini` and changed `__PORT__` and `__INSTALL_DIR__` inside it properly, and changed db_conn back to the default `sqlite` (and commenting out `database_engine_option_server_side_cursors` option), I manage to run a galaxy instance locally. 
![image](https://user-images.githubusercontent.com/9062727/43373668-2e9c1dda-9379-11e8-941a-6a0398b10daf.png)

Note that postgres didn't work as the machine I tested does not have postgres installed. Also there was a python dependency (`psycopg2-binary`) missing that's required to run `sqlalchemy` connected to a postgres. I had to run `.venv/bin/pip install psycopg2-binary` to pass around a python error and actually hit the wall of not having postgres running.
